### PR TITLE
fix: Correctly format index names

### PR DESCRIPTION
**Pull request by ericf-firebolt**
### Description

Updated logic in FireboltIndexConfig.parse() in impl.py to deal with situations where key columns have only a single value. Index names are now formatted properly.

### Checklist

- [x] I have run this code in development and it appears to resolve the stated issue.
- [x] This PR includes tests, or tests are not required/relevant for this PR.
- [x] I have updated `CHANGELOG.md` and added information about my change.
- [x] I have removed any print or log calls that were added for debugging.
- [x] If this PR requires a new PyPI release I have bumped the version number.
- [x] I have verified that this PR contains only code changes relevant to this PR.
- [x] If further integration tests are now passing I've edited firebolt.dbtspec to account for this.
- [x] I have pulled/merged from the main branch if there are merge conflicts.
- [x] After pulling/merging main I have run pytest on the included or updated firebolt.dbtspec.

### CODE REVIEW

File: dbt/adapters/firebolt/impl.py [hunk 1]
```diff
@@ -28,19 +28,32 @@ class FireboltIndexConfig(dbtClassMixin):
...
-        index type, relation name, key/join column, timestamp (unix & UTC)
+        index type, relation name, key/join columns, timestamp (unix & UTC)
         example index name: join_my_model_customer_id_1633504263.
         """
         now_unix = time.mktime(datetime.utcnow().timetuple())
-        spine_col = '_'.join(self.key_column if self.key_column else self.join_column)
+        # If column_names is a list with > 1 members, do a string join,
+        #  otherwise do not. We were getting index names like
+        # join__idx__emf_customers__f_i_r_s_t___n_a_m_e__165093112.
+        if self.key_column:
```
Comment (stepansergeevitch) [inline dbt/adapters/firebolt/impl.py:38]: I think we can simplify this part
Doing join on a list with a single element would result in that element
```
column_names = self.key_column or self.join_column
spine_col = '_'.join(column_names) if isinstance(column_names, list) else column_names
```

File: dbt/adapters/firebolt/impl.py [hunk 2]
```diff
@@ -28,19 +28,32 @@ class FireboltIndexConfig(dbtClassMixin):
...
+            spine_col = column_names[0]
         inputs = [
-            str(self.index_type),
-            'idx',
+            f'{self.index_type}_idx',
             relation.identifier,
             spine_col,
             str(int(now_unix)),
         ]
-        string = '__'.join(inputs)[:254]
```
Comment (stepansergeevitch) [inline dbt/adapters/firebolt/impl.py:43]: Why did we have a slice in the first place?

File: dbt/adapters/firebolt/impl.py (hunk 1 above)
Comment (stepansergeevitch) [reply dbt/adapters/firebolt/impl.py:38]: Also, why are these fields called `*_column`, when they might contain a list?

File: dbt/adapters/firebolt/impl.py (hunk 2 above)
Comment (ericf-firebolt) [reply dbt/adapters/firebolt/impl.py:43]: It was a legacy of a previous dev and the point wasn't clear to me.

File: dbt/adapters/firebolt/impl.py (hunk 1 above)
Comment (ericf-firebolt) [reply dbt/adapters/firebolt/impl.py:38]: Yeah, that's a _lot_ cleaner.

I didn't name those variables, I inherited them, but it's a good point. I've updated the names.

**Review by stepansergeevitch — APPROVED** [review]

File: dbt/adapters/firebolt/impl.py (hunk 2 above)
Comment (dataders) [reply dbt/adapters/firebolt/impl.py:43]: I think 256 is the max # of characters you can have for an index name...

**Pull request merged**